modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/extension/v1_0/ObjectRelationshipExtensionProvider.java

```diff
@@ -168,8 +195,54 @@ public boolean isApplicableExtension(long companyId, String className) {
 
 	@Override
 	public void setExtendedProperties(
-		long companyId, long userId, String className, Object entity,
-		Map<String, Serializable> extendedProperties) {
+			long companyId, long userId, String className, Object entity,
+			Map<String, Serializable> extendedProperties)
+		throws Exception {
+
+		ObjectDefinition objectDefinition = getObjectDefinition(
```

Comment:
I think we have to be careful with this `getObjectDefinition` method. It is not following the Liferay pattern of throwing an exception. It could return null, it is more of a `fetchObjectDefinition`
I say it because maybe the objectDefinition returned is null and ends in an NPE (it shouldn't be, but nothing warranty that it could not happen). 
I think that we should check for null.

Reply:
I agree with Luismi, we could rename the method to `fetchObjectDefinition`